### ISSUE TITLE
Data buffer not correct size.

### ISSUE THREAD

**Issue by daHiry** [PrismarineJS/mineflayer#666]
I'm not too sure how to proceed with my bot. The core mechanic of it is to mine down to bedrock using a special pick. When I give it a regular pickaxe it works perfect, but when I give it a 7x7 pickaxe, which mines a 7x7 radius around the block you break, I get "Data buffer not correct size". I've tried surrounding the code in a try catch but that doesn't seem to work either.

https://gyazo.com/67543bd41d3c38e253d35822a19cb85d

**Comment by rom1504** [comment]
Do you have the code of the plugin that create that 7x7 pickaxe ?
This is a completely non vanilla feature and as such we probably have to
adapt a few things to make it work.

**Comment by rom1504** [comment]
Also try to enable debug mode with DEBUG=minecraft-protocol node bot.js and
report back the packets just before the crash.

**Comment by daHiry** [comment]
The plugin is this plugin, https://www.spigotmc.org/resources/shockwavetools-massivecore-uuid.43854/
The packets, https://pastebin.com/jNsAtMU2

**Comment by rom1504** [comment]
Ah thanks. I think I know why it happens. The server sends a `groundUp: false` in the chunk packet and I think we don't handle that in prismarine-chunk.
I'll have a look

**Comment by rom1504** [comment]
Yeah it def doesn't handle it https://github.com/PrismarineJS/prismarine-chunk/blob/master/src/pc/1.9/chunk.js#L316
https://github.com/PrismarineJS/prismarine-chunk/blob/master/src/pc/1.8/chunk.js#L161

vanilla always send groundUp: true

I guess we need to add that feature in prismarine-chunk so that your server can work.

http://wiki.vg/Chunk_Format#Ground-up_continuous

that's the description of that property.

**Comment by rom1504** [comment]
opened an issue for it there https://github.com/PrismarineJS/prismarine-chunk/issues/43
